web/packages/teleport/src/components/TdpClientCanvas/TdpClientCanvas.tsx

```diff
@@ -98,6 +99,42 @@ function TdpClientCanvas(props: Props) {
     }
   }, [client, clientOnPngFrame]);
 
+  const previousCursor = useRef('auto');
+
+  useEffect(() => {
+    if (client && updatePointer) {
+      const canvas = canvasRef.current;
+      const updatePointer = (pointer: {
+        data: ImageData | boolean;
+        hotspot_x?: number;
+        hotspot_y?: number;
+      }) => {
+        if (typeof pointer.data === 'boolean') {
+          if (pointer.data) {
+            canvas.style.cursor = previousCursor.current;
+          } else {
+            previousCursor.current = canvas.style.cursor;
+            canvas.style.cursor = 'none';
+          }
+          return;
+        }
+        const cursor = document.createElement('canvas');
+        cursor.width = pointer.data.width;
+        cursor.height = pointer.data.height;
+        cursor.getContext('2d').putImageData(pointer.data, 0, 0);
```

Comment:
This probably doesn't change much in this case, but `pointer.data` returns a color space that we can pass to the canvas:
```ts
        cursor
          .getContext('2d', { colorSpace: pointer.data.colorSpace })
          .putImageData(pointer.data, 0, 0);
````